java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -767,4 +799,52 @@ public String getClientLibToken() {
   public String getQuotaProjectId() {
     return quotaProjectId;
   }
+
+  /**
+   * Returns the resolved host for the Service to connect to Google Cloud
+   *
+   * <p>The resolved host will be in `https://www.{serviceName}.{resolvedUniverseDomain}/` format.
+   * The resolvedUniverseDomain will be set to `googleapis.com` if universeDomain is null.
+   */
+  @InternalApi
+  public String getResolvedHost(String serviceName) {
+    if (universeDomain != null && universeDomain.isEmpty()) {
+      throw new IllegalArgumentException("Universe Domain cannot be empty");
```

Comment:
```suggestion
      throw new IllegalArgumentException("Universe Domain cannot be empty string");
```

Reply:
Guidance is to use `The universe domain cannot be empty`. I'll update that for consistency among languages.

Reply:
You could say there's specification for the exception message.